src/vm/database/marf.rs

```diff
@@ -328,6 +328,13 @@ impl MarfedKV {
         //    _if_ for some reason, we do want to be able to access that mined chain state in the future,
         //    we should probably commit the data to a different table which does not have uniqueness constraints.
         self.side_store.rollback(&self.chain_tip);
+        let _ = self.marf.commit_mined(will_move_to).map_err(|e| {
+            error!(
+                "Failed to commit to mined MARF block {}: {:?}",
+                &will_move_to, &e
+            );
+            panic!();
+        });
         self.marf
```

Comment:
Does this try to commit the MARF trie twice?

Reply:
Ah, looks like a merge artifact.

Reply:
Fixed